src/network/socket_wrapper.hpp

```diff
@@ -267,23 +267,38 @@ class TcpSocket {
   inline TcpSocket Accept() {
     SOCKET newfd = accept(sockfd_, NULL, NULL);
     if (newfd == INVALID_SOCKET) {
-      Log::Fatal("Socket accept error, code: %d", GetLastError());
+      int err_code = GetLastError();
+#if defined(_WIN32)
+      Log::Fatal("Socket accept error (code: %d)", err_code);
+#else
+      Log::Fatal("Socket accept error, %s (code: %d)", std::strerror(err_code), err_code);
```

Comment:
Shall we switch to a thread-safe version of `strerror`, (`strerror_r` for non-windows and `strerror_s` for windows)?

Reply:
@shiyu1994 sorry for the delay, I'm back to this.

Will that cause portability problems?

I saw in https://en.cppreference.com/w/c/string/byte/strerror that those functions will only work when `__STDC_LIB_EXT1__` is defined.

> As with all bounds-checked functions, strerror_s and strerrorlen_s are only guaranteed to be available if __STDC_LIB_EXT1__ is defined by the implementation and if the user defines __STDC_WANT_LIB_EXT1__ to the integer constant 1 before including string.h.

And when I researched a little bit about support for that, I found some posts that suggest that those aren't likely to be support by default in many setups.

* (Aug 2017) https://stackoverflow.com/a/45578649/3986677
* (Dec 2017) https://stackoverflow.com/questions/47867130/stdc-lib-ext1-availability-in-gcc-and-clang
* (Jun 2018) https://stackoverflow.com/questions/50724726/why-didnt-gcc-or-glibc-implement-s-functions
* (Dec 2020) https://stackoverflow.com/questions/65471315/scanf-s-is-not-included-in-c11
* (Dec 2020) https://stackoverflow.com/a/65471801/3986677

Reply:
I see. So maybe keep the current change is the best option.

Reply:
Ok, thanks very much for bringing it up! I learned new things reading about this.